File: Longest Subarray of 1's After Deleting One Element/longest-subarray-of-1s-after-deleting-one-element.py
class Solution(object):
    def longestSubarray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        curr = 0
        prev = 0
        max_len = 0

        for num in nums:
            if num == 1:
                curr += 1
            else:
                max_len = max(max_len, prev + curr)
                prev = curr
                curr = 0

        max_len = max(max_len, prev + curr)

        if max_len == len(nums):
            return max_len - 1

        return max_len

File: Longest Subarray of 1's After Deleting One Element/longest-subarray-of-1s-after-deleting-one-element.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int curr = 0, prev = 0, maxlen = 0;
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] == 1)
                curr++;
            else {
                maxlen = max(maxlen, prev + curr);
                prev = curr;
                curr = 0;
            }
        }
        maxlen = max(maxlen, prev + curr);
        if (maxlen == nums.size())
            return maxlen - 1;
        return maxlen;
    }
};